src/components/edit-user.html

```diff
@@ -37,31 +34,93 @@
       }
     </style>
     <h3>Edit User Component</h3>
-    <form>
-      <fieldset>
-        <div>
-          <label for="firstName">First name:</label>
-          <input type="text" name="firstName" id="firstName">
-        </div>
-        <div>
-          <label for="lastName">Last name:</label>
-          <input type="text" name="lastName" id="lastName">
-        </div>
-        <div>
-          <label for="phone">Phone:</label>
-          <input type="text" name="phone" id="phone">
-        </div>
-        <div>
-          <label for="email">Email:</label>
-          <input type="text" name="email" id="email">
-        </div>
-        <button type="submit">Save Edit</button>
-      </fieldset>
-    </form>
+    <template is="dom-if" if="[[editSelected]]">
+      <table>
+        <tr>
+          <td>ID :</td>
+          <td>[[user.id]]</td>
+        </tr>
+        <tr>
+          <td><label for="name">Name :</label></td>
+          <td>[[user.firstName]] [[user.lastName]]</td>
+        </tr>
+        <tr>
+          <td><label for="phone">Phone :</label></td>
+          <td>[[user.phoneNumber]]</td>
+        </tr>
+        <tr>
+          <td><label for="email">Email :</label></td>
+          <td>[[user.email]]</td>
+        </tr>
+        <tr>
+          <td><button type="button" on-click="editUserProfile">[[editButtonText(editSelected)]]</button></td>
+        </tr>
+      </table>
+    </template>
+    <template is="dom-if" if="[[!editSelected]]">
+      <table>
+        <tr>
+          <td><label for="firstName">First name :</label></td>
+          <td><input type="text" name="firstName" id="firstName" value="{{user.firstName}}" on-input="updateFirstName"></td>
```

Comment:
So you're already using two way data binding here do you even need to use the on-input listener here for it to work?

If so you could also do something like `value="{{user.firstName::input}}"` which is basically saying, "whenever there is an on-input event, update user.firstName."

For more in depth on that the bottom of this page:

https://polymer-library.polymer-project.org/1.0/docs/devguide/data-binding

Reply:
@kddrake  went over that way of solving this with the `::input` and mentioned that it's not the prefered way.  I believe it had something to do with the value having to immediately change rather than save and have the changes made.  Either way I think it just depends on the circumstances. For the record I know both ways now.